crates/proof-of-sql/src/sql/proof/proof_plan.rs

```diff
@@ -36,16 +36,14 @@ pub trait ProofPlan: Debug + Send + Sync + ProverEvaluate {
 
 #[enum_dispatch::enum_dispatch(DynProofPlan)]
 pub trait ProverEvaluate {
-    /// Evaluate the query and return the result.
-    fn result_evaluate<'a, S: Scalar>(
+    /// Evaluate the query, modify `FirstRoundBuilder` and return the result.
+    fn first_round_evaluate<'a, S: Scalar>(
         &self,
+        builder: &mut FirstRoundBuilder,
         alloc: &'a Bump,
         table_map: &IndexMap<TableRef, Table<'a, S>>,
     ) -> (Table<'a, S>, Vec<usize>);
```

Comment:
NIT:
One of the motivations for doing this was so that the lengths could be part of the builder rather than an output.